File: shared/serialization/src/commonTest/kotlin/kmp/benchmarks/serialization/kotlinxserialization/KotlinxSerializationPrettyPrintedTest.kt
package kmp.benchmarks.serialization.kotlinxserialization

import kmp.benchmarks.serialization.Resources
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class KotlinxSerializationPrettyPrintedTest {
    @Test
    fun decodeLargeListFromString() {
        val data = KotlinxSerializationPrettyPrinted.decodeLargeListFromString(Resources.PrettyPrinted.largeListJsonString)
        assertTrue(data.isNotEmpty())
    }

//    @Test
//    fun encodeLargeListToString() {
//        val data = KotlinxSerializationPrettyPrinted.decodeLargeListFromString(Resources.PrettyPrinted.largeListJsonString)
//        assertEquals(Resources.PrettyPrinted.largeListJsonString, KotlinxSerializationPrettyPrinted.encodeLargeListToString(data))
//    }

    @Test
    fun decodeMacOsReleasesFromString() {
        val data = KotlinxSerializationPrettyPrinted.decodeMacOsReleasesFromString(Resources.PrettyPrinted.macosReleasesJsonString)
        assertTrue(data.versions.isNotEmpty())
    }

    @Test
    fun encodeMacOsReleasesToString() {
        val data = KotlinxSerializationPrettyPrinted.decodeMacOsReleasesFromString(Resources.PrettyPrinted.macosReleasesJsonString)
        assertEquals(Resources.PrettyPrinted.macosReleasesJsonString, KotlinxSerializationPrettyPrinted.encodeMacOsReleasesToString(data))
    }

    @Test
    fun decodePolymorphicGeoFromString() {
        val data = KotlinxSerializationPrettyPrinted.decodePolymorphicGeoFromString(Resources.PrettyPrinted.polymorphicGeoJsonString)
        assertTrue(data.isNotEmpty())
    }

    @Test
    fun encodePolymorphicGeoToString() {
        val data = KotlinxSerializationPrettyPrinted.decodePolymorphicGeoFromString(Resources.PrettyPrinted.polymorphicGeoJsonString)
        assertEquals(Resources.PrettyPrinted.polymorphicGeoJsonString, KotlinxSerializationPrettyPrinted.encodePolymorphicGeoToString(data))
    }

    @Test
    fun decodePolymorphicHtmlFromString() {
        val data = KotlinxSerializationPrettyPrinted.decodePolymorphicHtmlFromString(Resources.PrettyPrinted.polymorphicHtmlJsonString)
        assertTrue(data.isNotEmpty())
    }

    @Test
    fun encodePolymorphicHtmlToString() {
        val data = KotlinxSerializationPrettyPrinted.decodePolymorphicHtmlFromString(Resources.PrettyPrinted.polymorphicHtmlJsonString)
        assertEquals(Resources.PrettyPrinted.polymorphicHtmlJsonString, KotlinxSerializationPrettyPrinted.encodePolymorphicHtmlToString(data))
    }

    @Test
    fun decodeUserProfileFromString() {
        val data = KotlinxSerializationPrettyPrinted.decodeUserProfileFromString(Resources.PrettyPrinted.userProfileJsonString)
        assertTrue(data.personal.firstName.isNotEmpty())
    }

    @Test
    fun encodeUserProfileToString() {
        val data = KotlinxSerializationPrettyPrinted.decodeUserProfileFromString(Resources.PrettyPrinted.userProfileJsonString)
        assertEquals(Resources.PrettyPrinted.userProfileJsonString, KotlinxSerializationPrettyPrinted.encodeUserProfileToString(data))
    }
}

File: app-ios/multiplatformTests/serialization/KotlinxSerializationPrettyPrintedTest.swift
//
//  KotlinxSerializationTests.swift
//  multiplatformTests
//
//  Created by Sebastian Owodzin on 12/01/2024.
//  Copyright © 2024 kmp. All rights reserved.
//

import BenchmarksShared
import XCTest

final class KotlinxSerializationPrettyPrintedTests: XCTestCase {

    func testDecodeLargeListFromString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodeLargeListFromString(jsonString: Resources.PrettyPrinted.largeListJsonString)
        XCTAssertTrue(!data.isEmpty)
    }
    
//    func testEncodeLargeListToString() throws {
//        let data = try! KotlinxSerializationPrettyPrinted().decodeLargeListFromString(jsonString: Resources.PrettyPrinted.largeListJsonString)
//        XCTAssertEqual(Resources.PrettyPrinted.largeListJsonString, try! KotlinxSerializationPrettyPrinted().encodeLargeListToString(value: data))
//    }
    
    func testDecodeMacOsReleasesFromString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodeMacOsReleasesFromString(jsonString: Resources.PrettyPrinted.macosReleasesJsonString)
        XCTAssertTrue(!data.versions.isEmpty)
    }
    
    func testEncodeMacOsReleasesToString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodeMacOsReleasesFromString(jsonString: Resources.PrettyPrinted.macosReleasesJsonString)
        XCTAssertEqual(Resources.PrettyPrinted.macosReleasesJsonString, try! KotlinxSerializationPrettyPrinted().encodeMacOsReleasesToString(value: data))
    }
    
    func testDecodePolymorphicGeoFromString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodePolymorphicGeoFromString(jsonString: Resources.PrettyPrinted.polymorphicGeoJsonString)
        XCTAssertTrue(!data.isEmpty)
    }
    
    func testEncodePolymorphicGeoToString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodePolymorphicGeoFromString(jsonString: Resources.PrettyPrinted.polymorphicGeoJsonString)
        XCTAssertEqual(Resources.PrettyPrinted.polymorphicGeoJsonString, try! KotlinxSerializationPrettyPrinted().encodePolymorphicGeoToString(value: data))
    }
    
    func testDecodePolymorphicHtmlFromString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodePolymorphicHtmlFromString(jsonString: Resources.PrettyPrinted.polymorphicHtmlJsonString)
        XCTAssertTrue(!data.isEmpty)
    }
    
    func testEncodePolymorphicHtmlToString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodePolymorphicHtmlFromString(jsonString: Resources.PrettyPrinted.polymorphicHtmlJsonString)
        XCTAssertEqual(Resources.PrettyPrinted.polymorphicHtmlJsonString, try! KotlinxSerializationPrettyPrinted().encodePolymorphicHtmlToString(value: data))
    }
    
    func testDecodeUserProfileFromString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodeUserProfileFromString(jsonString: Resources.PrettyPrinted.userProfileJsonString)
        XCTAssertTrue(!data.personal.firstName.isEmpty)
    }
    
    func testEncodeUserProfileToString() throws {
        let data = try! KotlinxSerializationPrettyPrinted().decodeUserProfileFromString(jsonString: Resources.PrettyPrinted.userProfileJsonString)
        XCTAssertEqual(Resources.PrettyPrinted.userProfileJsonString, try! KotlinxSerializationPrettyPrinted().encodeUserProfileToString(value: data))
    }
}
